numba/tests/test_np_functions.py

```diff
@@ -4965,6 +4969,128 @@ def test_cross2d_exceptions(self):
             str(raises.exception)
         )
 
+    def test_trim_zeros(self):
+
+        def arrays():
+            yield np.array([])
+            yield np.zeros(5)
+            yield np.zeros(1)
+            yield np.array([1, 2, 3])
+            yield np.array([0, 1, 2, 3])
+            yield np.array([0., 1., 2., np.nan, 0.])
+            yield np.array(['0', 'Hello', 'world'])
+
+        def explicit_trim():
+            yield np.array([0, 1, 2, 0, 0]), 'FB'
+            yield np.array([0, 1, 2]), 'B'
+            yield np.array([np.nan, 0., 1.2, 2.3, 0.]), 'b'
+            yield np.array([0, 0, 1, 2, 5]), 'f'
+            yield np.array([0, 1, 2, 0]), 'abf'
+            yield np.array([0, 4, 0]), 'd'
+            yield np.array(['\0', '1', '2']), 'f'
+
+        pyfunc = np_trim_zeros
+        cfunc = jit(nopython=True)(pyfunc)
+
+        for arr in arrays():
+            expected = pyfunc(arr)
+            got = cfunc(arr)
+            self.assertPreciseEqual(expected, got)
+
+        for arr, trim in explicit_trim():
+            expected = pyfunc(arr, trim)
+            got = cfunc(arr, trim)
+            self.assertPreciseEqual(expected, got)
+
+    def test_trim_zeros_NumPy(self):
```

Comment:
It's a tiny bit odd to switch into camel case for half of the test name:

```suggestion
    def test_trim_zeros_numpy(self):
```